### PR TITLE
LL-7025 LL-7059 Language discoverability

### PR DESCRIPTION
**Pull request by tlaforge-ledger**
#### **EDIT: (13/01) @ofreyssinet-ledger:**
new changes explained in this comment: https://github.com/LedgerHQ/ledger-live-mobile/pull/1926#issuecomment-1012268761

TL;DR:
- default initial language (for new users) to system language if that language is supported [LL-7059]
- some refactoring to change where the source of truth for i18next is and where it gets updated
___
#### Initial PR comment:

- use crypto-js to replace react-native-crypto
- add pushed languages
- removed fr from preferred languages
- add checker for language availability
- move languages list
- add translation keys
- add language logo
- checker component
- add component on default screen to use hooks like useTheme

### Type

New Feature

### Context

[LL-7025]
[LL-7059]
[LL-8921]

### Parts of the app affected / Test plan

On startup, on default screen (portfolio), it add a modal first time you start with another language than default on your mobile.
The language must be part of translates trusted list.
It give you the possibility to switch to this language. If you say yes, or not, we handle the answer and remember it so you will no longer get the info.

[LL-7025]: https://ledgerhq.atlassian.net/browse/LL-7025
[LL-7059]: https://ledgerhq.atlassian.net/browse/LL-7059

[LL-8921]: https://ledgerhq.atlassian.net/browse/LL-8921?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -219,7 +220,7 @@
     "zlib": "browserify-zlib",
     "console": "console-browserify",
     "constants": "constants-browserify",
-    "crypto": "react-native-crypto",
+    "crypto": "crypto-js",
```
Comment (LFBarreto) [inline package.json:223]: could you remove this from this PR it goes a bit out of the scope of the feature :+1: 
@valpinkman @MortalKastor to address the update needed and create a separate PR for it

File: src/components/CheckLanguageAvailability.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          footer={
+            <View>
+              <Button
+                type="primary"
+                event={`Discoverability - Switch - ${osLanguage}`}
+                eventProperties={{ language: osLanguage }}
+                title={
+                  <>
+                    <Trans i18nKey="systemLanguageAvailable.switchButton" />
+                    <Text> {languages[osLanguage]}</Text>
```
Comment (LFBarreto) [inline src/components/CheckLanguageAvailability.js:85]: you could maybe use the language in the i18n template in that case :thinking: 
```
<Trans i18nKey="systemLanguageAvailable.switchButton" values={{language: languages[osLanguage] }} />
... en.app.json
"switchButton":  "switch to {{language}}",
...

File: src/context/Locale.js [hunk 3]
```diff
@@ -89,3 +89,34 @@ export default function LocaleProvider({ children }: Props) {
...
+  const [checked, setChecked] = useState(false);
+
+  const accept = useCallback(() => {
+    answerLanguageAvailable().then(() => {
+      setChecked(true);
+    });
+  }, []);
+
+  useEffect(() => {
+    hasAnsweredLanguageAvailable().then(answerLanguageAvailable);
```
Comment (LFBarreto) [inline src/context/Locale.js:118]: did you mean to 
 ```
 hasAnsweredLanguageAvailable().then(setChecked)
 ```
 instead :thinking:

**Review by LFBarreto — CHANGES_REQUESTED** [review]
Some minor changes required :+1:

File: package.json (hunk 1 above)
Comment (tlaforge-ledger) [reply package.json:223]: ok for me, do I need a Jira Ticket ? Which squad ? or only PR with @valpinkman and @MortalKastor mentionned ?

**Review by LFBarreto — DISMISSED** [review]

File: package.json [hunk 4]
```diff
@@ -94,6 +94,7 @@
     "color": "^3.1.3",
     "console-browserify": "^1.2.0",
     "constants-browserify": "^1.0.0",
+    "crypto-js": "3.1.9-1",
```
Comment (gre) [inline package.json:97]: unclear why this is related to this PR.

**Review by gre — DISMISSED** [review]

**Review by valpinkman — APPROVED** [review]

**Comment by ofreyssinet-ledger** [comment]
Fixed the typo as requested in [LL-8921]

**Review by valpinkman — APPROVED** [review]

**Comment by ofreyssinet-ledger** [comment]
Just rebased and force pushed.
I added some commits with the main purpose of having the system's language used as a default for a new user in case that language is supported. [LL-7059]

I also did some refactoring regarding where is the source of truth for the language locale, it is now in the store's `setting.locale` field and i18next gets updated with that value when it changes.

This `setting.locale` field is the one being initialised to the system language if it's one of the supported languages.

I need this refactoring for another PR (https://github.com/LedgerHQ/ledger-live-mobile/pull/1941) regarding the new "region" distinct locale that we will have, as it needs a simple initialisation logic based on the language's initialisation logic, and it's a value that lives in the store, hence why I needed the logic of the language initialisation to be done there. I hope that makes sense, if not we can discuss it.

File: src/components/CheckLanguageAvailability.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    modalOpened &&
+    !hasAnswered &&
+    currAppLanguage !== osLanguage &&
+    pushedLanguages.includes(osLanguage);
+
+  if (!toShow) {
+    return null;
+  }
+
+  const onRequestClose = () => {
```
Comment (LFBarreto) [inline src/components/CheckLanguageAvailability.js:40]: maybe use Callback here to not re-render these callbacks :thinking: 
the show or not show doesn't impact it so you can still do hooks and have a condition afterwards

**Review by LFBarreto — APPROVED** [review]

File: src/components/CheckLanguageAvailability.js (hunk 5 above)
Comment (ofreyssinet-ledger) [reply src/components/CheckLanguageAvailability.js]: done https://github.com/LedgerHQ/ledger-live-mobile/pull/1926/commits/2b2ebb9aa33a2836d9e2181f28084af19b01debd

**Review by LFBarreto — APPROVED** [review]

**Pull request merged**